selftests/functional/test_interrupt.py

```diff
@@ -176,13 +208,159 @@ def wait_until_no_goodtest():
 
             return len(good_test_processes) == 0
 
-        wait.wait_for(wait_until_no_goodtest, timeout=2)
+        if not wait.wait_for(wait_until_no_goodtest, timeout=2):
+            self.fail('Avocado left processes behind.')
+
+        output = proc.communicate()[0]
         # Make sure the Killing test subprocess message is not there
-        self.assertNotIn('Killing test subprocess', proc.get_output())
+        self.assertNotIn('Killing test subprocess', output)
         # Make sure the Interrupted requested sentence is there
         self.assertIn('Interrupt requested. Waiting 2 seconds for test to '
-                      'finish (ignoring new Ctrl+C until then)',
-                      proc.get_output())
+                      'finish (ignoring new Ctrl+C until then)', output)
+
+    @unittest.skipIf(int(os.environ.get("AVOCADO_CHECK_LEVEL", 0)) < 2,
+                     "Skipping test that take a long time to run, are "
+                     "resource intensive or time sensitve")
+    def test_badly_behaved_sigterm(self):
```

Comment:
This test can use most of the `test_badly_behaved_sigint` methods, this way it's really hard to follow what is different here and there. Could you please refactor it? (this way it'll become pain to maintain it)